typings.d.ts

```diff
@@ -110,29 +146,101 @@ declare module 'resin-components' {
 		setRules(rules: FilterRule[]): void;
 	}
 
-	class Filters extends RenderableElementWithProps<FiltersProps, any> {
-		refs: any;
-		render(): JSX.Element | null;
+	class Filters extends RenderableElementWithProps<FiltersProps, any> {}
+
+	class Fixed extends RenderableElementWithProps<DefaultProps, any> {}
+
+	interface FlexProps extends BoxProps {
+		align?: string | string[];
+		justify?: string | string[];
+		direction?: string | string[];
+		wrap?: boolean | boolean[] | string;
+		column?: boolean;
 	}
 
-	interface ProgressBarProps extends DefaultProps {
-		value: number;
-		info?: boolean;
+	class Flex extends RenderableElementWithProps<FlexProps, any> {}
+
+	namespace Heading {
+		class h4 extends RenderableElementWithProps<FlexProps, any> {}
+	}
+
+	interface LinkProps
+		extends DefaultProps,
+			React.AnchorHTMLAttributes<HTMLElement> {
+		blank?: boolean;
+		disabled?: boolean;
 	}
 
-	class ProgressBar extends RenderableElementWithProps<ProgressBarProps, any> {
-		refs: any;
-		render(): JSX.Element | null;
+	class Link extends RenderableElementWithProps<LinkProps, any> {}
+
+	class Input extends RenderableElementWithProps<DefaultProps, any> {}
+
+	class Image extends RenderableElementWithProps<
+		DefaultProps & { src: string },
+		any
+	> {}
+
+	class Modal extends RenderableElementWithProps<
+		DefaultProps & {
+			title: string;
```

Comment:
Heads-up, this is also part of the `HTMLAttributes`, but we probably don't have to do something for that.